lib/fuser_basic.h

```diff
@@ -79,13 +93,33 @@ struct BasicGateFuser final {
         gates_lat[k].reserve(128);
       }
 
+      auto prev_time = gates[gate_index].time;
+
       // Fill gates_seq and gates_lat in.
       for (; gate_index < gates.size(); ++gate_index) {
         const auto& gate = gates[gate_index];
 
-        if (gates[gate_index].time > times_to_split_at[l]) break;
+        if (gate.time > times[l]) break;
+
+        if (gate.time < prev_time) {
+          // This function assumes that gate times are ordered.
+          // Just stop sielently if this is not the case.
+          // TODO: report an error here.
```

Comment:
`sielently` -> `silently`

Also: can the TODO be addressed in this PR?

Reply:
Done.